Add post on GitHub pro tips

mriddle opened the pull request:
Sharing some pro tips that some of us might not know. 
Used for a lunch and learn in the US office

mriddle committed:
Add post on GitHub pro tips

vinbarnes commented:
[rendered markdown](https://github.com/lonelyplanet/blog/blob/mr-github-pro-tips/_posts/2014-06-17-GitHub-Tips-n-Tricks.md)

leejones committed:
Fix typo

leejones commented on _posts/2014-06-17-GitHub-Tips-n-Tricks.md:
```diff
@@ -0,0 +1,82 @@
+
+`git commit -m "Roll out new UI from lonelyplanet/rizzo#782"`
+
+Late commit relating to a merged PR (will show up in the discussion)
```
@mriddle what did you mean by “Late commit”?

mriddle replied:
@leejones Committing code that's relevant to a pull request which has already been merged.

vinbarnes committed:
tweak language

vinbarnes committed:
add intro and links to search tip

vinbarnes committed:
add notes to compare tip and consistent look/feel

vinbarnes commented:
@mriddle made some tweaks to get the look/feel more consistent, etc. If I took editorial liberties that aren't in keeping with your vision for the article feel free to revert.

vinbarnes committed:
fix typo

vinbarnes commented:
Minor preference for having or own slothy reply gif rather than using GH's. :grin:

mriddle committed:
Add more search examples

Not sure how many people know you can narrow your search down by location :)

mriddle committed:
moar sloth

vinbarnes commented:
5810a77 :zap:

vinbarnes committed:
typo

TimMoore commented on _posts/2014-06-17-GitHub-Tips-n-Tricks.md:
```diff
@@ -0,0 +1,91 @@
+
+**Find sensitive information**
+
+* [`"PRIVATE KEY" in:file path:.ssh/id_rsa`](https://github.com/search?q=%22PRIVATE+KEY%22+in%3Afile+path%3A.ssh%2Fid_rsa&type=Code&ref=searchresults)
```
I know it's already public, but I'm not so sure it's good form to "out" people this way.

Though, https://github.com/johndouthat/mydotfiles/blob/81b657398c71c33c6b7459c5aa8c2f070adadca9/.ssh/id_rsa I LOL'ed

mriddle replied:
Great point, I was trying to convey how easy it is and that other GitHub users should be aware.
I'm going to add the user param so just that file shows up instead, also made me laugh.

TimMoore commented on _posts/2014-06-17-GitHub-Tips-n-Tricks.md:
```diff
@@ -0,0 +1,91 @@
+
+## Code comparison
+
+You may already be familiar with using [http://github.com/<org>/<repo>/compare]() to create a pull-request from a branch, but it's also super useful for filtering commits.
```
I think you need to escape `<org>` and `<repo>` here. At least in the rendered view of the PR, they're invisible.

Putting backticks around it should work.

TimMoore committed:
Backtick-escape all raw URLs in the text.

TimMoore committed:
Self-link the example compare URL.

It doesn't actually go anywhere valid, but people can edit the URL with their own org and repo name.

Alternately, we could unlink it entirely, but the way it was before, it was just an empty link back to this post.

TimMoore commented:
Nice post! I tweaked a couple of things, too, and to echo Kevin: feel free to revert if you don't like any of them (although I think I just changed his changes).

leejones commented:
:+1: nice work @mriddle :sparkles:

mriddle committed:
Update sensitive search examples

Don't want to call people out on it, just make users aware that the internet knows all.

mriddle commented:
Thanks for the feedback team! :sparkling_heart:
